WhoIsThat/WhoIsThat/ViewModels/LoginViewModel.cs

```diff
@@ -1,25 +1,115 @@
+
         public INavigation Navigation { get; set; }
 
-        public LoginViewModel()
+        public event PropertyChangedEventHandler PropertyChanged;
+
+                
+        private MediaFile takenPhoto { get; set; }
+        private ImageObject personObject;
+
+        public ImageObject PersonObject
+        {
+            get
+            {
+                return personObject;
+            }
+            set
+            {
+                personObject = value;
+                OnPropertyChanged();
+            }
+        }
+
+        public async void TakePhoto()
         {
-            NavigateHomePageCommand = new Command(NavigateToHomePage);
+            //Checking for camera permissions
+            bool cameraPermission = await PermissionHandler.CheckForCameraPermission();
+            if (!cameraPermission)
+                await CrossPermissions.Current.RequestPermissionsAsync(Permission.Camera);
+
+            //Checking for storage permissions
+            bool storagePermission = await PermissionHandler.CheckForCameraPermission();
+            if (!storagePermission)
+                await CrossPermissions.Current.RequestPermissionsAsync(Permission.Storage);
+
+            //Taking photo and storing it in MediaFile variable 'takenPhoto'
+
+            takenPhoto = await TakingPhotoHandler.TakePhoto();
+
+            //Save taken photo to Azure cloud for recognition, later on it is deleted
+
+            //await CloudStorageService.SaveBlockBlob(takenPhoto);
+        }
+
+        public async void SavePerson()
+        { 
+            //Checks if all required fields are filled
+
+            /*if(PersonObject.PersonFirstName == null || PersonObject.PersonLastName == null || PersonObject.DescriptiveSentence == null || takenPhoto == null)
+            {
+                return;
+            }*/
+
+            //If so, stores his information in the db
+            PersonObject.Id = 1;
+            PersonObject.ImageName = PersonObject.PersonFirstName + PersonObject.PersonLastName + ".jpg";
+            /*await CloudStorageService.SaveBlockBlob(takenPhoto,PersonObject.ImageName);
+
+            //Information stored and the stored user is returned with a newly generated ID
+            ImageObject person = await _restService.CreateImageObject(PersonObject);
+
+            
+            */
+            //Saves the 'state' of user as registered
+            if (Application.Current.Properties.ContainsKey("UserRegistered"))
+            {
+                Application.Current.Properties.Remove("UserRegistered");
+            }
+            Application.Current.Properties.Add("UserRegistered", true);
+            await Application.Current.SavePropertiesAsync();
         }
 
         public async void NavigateToHomePage()
         {
             await Application.Current.MainPage.Navigation.PushAsync(new HomePage(new HomeViewModel()));
         }
+        public void OnPropertyChanged([CallerMemberName] string propertiesName = "")
```

Comment:
Is this really better than it was before?

@tomasmuzas maybe you could check this feature out

Reply:
![screenshot_1](https://user-images.githubusercontent.com/30218599/47881083-d13b0200-de2d-11e8-8120-0619bc875ec7.png)
There are a lot of ways to implement OnPropertyChanged method and they don't really differ much.  This implementation accepts arguments (calling OnPropertyChanged("someProperty") is acceptable), but can also take the callers name as an argument. Similar would be OnPropertyChanged(nameof(value)) in this case. 

However, i'm not even sure this feature is required here, even though all sources suggest it is. I tried filling the fields with and without triggering this method, and it turns out fine either way.